src/components/Profile/PriorityProposalsBox.tsx

```diff
@@ -0,0 +1,173 @@
+  priorityProposals?: PriorityProposal[],
+  lowerAddress?: string | null
+) {
+  return votes && priorityProposals && lowerAddress
+    ? priorityProposals?.filter((proposal) => {
+        const hasVotedOnMain = votes && lowerAddress && votes[proposal.id] && !!votes[proposal.id][lowerAddress]
+        const hasVotedOnLinked =
+          proposal.linked_proposals_data &&
+          proposal.linked_proposals_data.some(
+            (linkedProposal) => votes[linkedProposal.id] && !!votes[linkedProposal.id][lowerAddress]
+          )
+        const hasAuthoredBid =
+          proposal.unpublished_bids_data &&
+          proposal.unpublished_bids_data.some((linkedBid) => isSameAddress(linkedBid.author_address, lowerAddress))
+
+        return !hasVotedOnMain && !hasVotedOnLinked && !hasAuthoredBid
+      })
+    : priorityProposals
+}
+
+function PriorityProposalsBox({ address, collapsible = false }: Props) {
+  const t = useFormatMessage()
+  const isTabletAndBelow = useTabletAndBelowMediaQuery()
+  const lowerAddress = address?.toLowerCase()
+  const { priorityProposals, isLoading } = usePriorityProposals(lowerAddress)
+  const proposalIds =
+    priorityProposals?.reduce((acc: string[], priorityProposal) => {
+      acc.push(priorityProposal.id)
+      if (priorityProposal.linked_proposals_data && priorityProposal.linked_proposals_data.length > 0) {
+        priorityProposal.linked_proposals_data.map((linkedProposal) => acc.push(linkedProposal.id))
+      }
+      return acc
+    }, []) || []
+
+  const { votes, isLoadingVotes } = useProposalsCachedVotes(proposalIds || [])
+  const [displayedProposals, setDisplayedProposals] = useState(
+    getDisplayedProposals(votes, priorityProposals, lowerAddress)
+  )
+
+  useEffect(() => {
+    setDisplayedProposals(getDisplayedProposals(votes, priorityProposals, lowerAddress))
+  }, [isLoadingVotes, lowerAddress, priorityProposals, votes])
+
+  const [displayedProposalsAmount, setDisplayedProposalsAmount] = useState(PROPOSALS_PER_PAGE)
+  const hasMoreProposals = displayedProposals && displayedProposals.length > PROPOSALS_PER_PAGE
+  const showViewMoreButton = hasMoreProposals && displayedProposalsAmount < displayedProposals.length
+  const showViewLessButton = hasMoreProposals && displayedProposalsAmount >= displayedProposals.length
+
+  const handleViewMore = () => {
+    if (displayedProposals) setDisplayedProposalsAmount(displayedProposals.length)
+  }
+
+  const handleViewLess = () => {
+    if (displayedProposals) setDisplayedProposalsAmount(PROPOSALS_PER_PAGE)
+  }
+
+  return isLoading || (!isLoading && priorityProposals && priorityProposals.length === 0) ? null : (
+    <div className="ProposalsPage__Priority">
+      {collapsible ? (
+        <ActionBox
+          title={
+            <span className="PriorityProposalsBox__Title">
+              <Counter gray count={displayedProposals?.length} />
+              {t('component.priority_proposals.title')}
+            </span>
```

Comment:
This can be `<PriorityProposalsBoxTitle counter={...} counterColor={...} />` that you can use here and in line 144.